examples/erc721/lib.rs

```diff
@@ -325,19 +348,14 @@ mod erc721 {
                 operator: to,
                 approved,
             });
-            if self.approved_for_all(caller, to) {
-                let status = self
-                    .operator_approvals
-                    .get_mut(&(caller, to))
-                    .ok_or(Error::CannotFetchValue)?;
-                *status = approved;
-                Ok(())
+
+            if approved {
+                self.operator_approvals.insert((&caller, &to), &approved);
             } else {
-                match self.operator_approvals.insert((caller, to), approved) {
-                    Some(_) => Err(Error::CannotInsert),
-                    None => Ok(()),
-                }
+                self.operator_approvals.remove((&caller, &to));
```

Comment:
Do we still need the `bool` as the value in the mapping after this change? Couldn't it be just `()` now that we only check for existence?